File: copl/com.elster.jupiter.webservices.extjs/src/main/web/js/webservices/src/store/Roles.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Wss.store.Roles', {
    extend: 'Ext.data.Store',
    model: 'Wss.model.Role',
    proxy: {
        type: 'rest',
        url: '/api/ws/fields/roles',
        timeout: 120000,
        reader: {
            type: 'json',
            root: 'roles'
        }
    }
});

File: copl/com.elster.jupiter.public.rest.util/src/main/java/com/elster/jupiter/rest/api/util/Roles.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.rest.api.util;

public enum Roles {
    DEVELOPER("Developer", "Grants access to public rest api");

    private String role;
    private String description;

    Roles(String r, String d) {
        role = r;
        description = d;
    }

    public String value() {
        return role;
    }

    public String description() {
        return description;
    }
}
